### PR TITLE
Remove checkstyle suppressions on Config and TestProperties #5478

### LINKED ISSUE #5478: Remove checkstyle suppressions on Config and TestProperties
Currently, `MemberName|StaticVariableName` is suppressed for `TestProperties` and `Config` in `teammates-checkstyle-suppressions.xml`. We should remove the suppressions and fix the violations on Config and TestProperties.

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #5478.

Not sure if this was what you had in mind, @wkurniawan07. 

Treating the properties as static fields is consistent with google app engine's [SystemProperty](https://cloud.google.com/appengine/docs/java/javadoc/com/google/appengine/api/utils/SystemProperty#class-systemproperty) class.

### CODE REVIEW

File: src/test/java/teammates/test/driver/HtmlHelper.java [hunk 1]
```diff
@@ -402,16 +404,16 @@ private static String replaceUnpredictableValuesWithPlaceholders(String content)
...
-                      .replace(Const.SystemParams.getjQueryUiFilePath(TP.isDevServer()), "${lib.path}/jquery-ui.min.js")
+                      .replace(Const.SystemParams.getjQueryUiFilePath(TestProperties.isDevServer()), "${lib.path}/jquery-ui.min.js")
                       // admin footer, test institute section
                       .replaceAll("(?s)<div( class=\"col-md-8\"| id=\"adminInstitute\"){2}>"
                                               + REGEX_ADMIN_INSTITUTE_FOOTER + "</div>",
                                   "\\${admin\\.institute}")
                       // top HTML tag with xmlns defined
                       // TODO check if this is necessary
-                      .replace("<html xmlns=\"http://www.w3.org/1999/xhtml\">", "<html>")
+                      .replace("<html xmlns=\"htTestProperties://www.w3.org/1999/xhtml\">", "<html>")
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/driver/HtmlHelper.java]: I don't think restarting the build helps if this is still here... :joy:

Comment (kanghj) [inline src/test/java/teammates/test/driver/HtmlHelper.java]: Oops, thought I committed already

**Comment by wkurniawan07** [comment]
I was thinking of making the fields in `TestProperties` just final, not static, so we don't have to change all the `inst()`, but this is clearly the better solution.

File: src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java [hunk 2]
```diff
@@ -257,19 +257,19 @@ private void testCreateInstructorAction() throws Exception {
 
         ______TS("action success: course is accessible for newly joined instructor as student");
         //in staging server, the student account uses the hardcoded email above, so this can only be test on dev server
-        if (!TestProperties.inst().TEAMMATES_URL.contains("local")) {
+        if (!TestProperties.TEAMMATES_URL.contains("local")) {
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: Pretty sure this is trying to say `TestProperties.isDevServer()`.

File: src/test/java/teammates/test/driver/TestProperties.java [hunk 3]
```diff
@@ -86,20 +82,16 @@ private TestProperties() {
...
         }
     }
-
-    public static TestProperties inst() {
-        if (instance == null) {
-            instance = new TestProperties();
-        }
-        return instance;
+    
+    private TestProperties() {
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/driver/TestProperties.java]: Add a comment on the empty constructor, like `Config`.

**Comment by wkurniawan07** [comment]
And when #5486 is merged in, you can remove the suppression filters for these two files.

File: src/test/java/teammates/test/driver/TestProperties.java [hunk 4]
```diff
@@ -133,20 +125,22 @@ public void verifyReadyForGodMode() {
...
             return false;
         }
-        return inst().TEST_STUDENT2_ACCOUNT.equals("charlie.tmms." + uniqueId)
-            && inst().TEST_INSTRUCTOR_ACCOUNT.equals("teammates.coord." + uniqueId)
-            && inst().TEST_ADMIN_ACCOUNT.equals("yourGoogleId." + uniqueId);
+        
+        boolean isSecondStudentAccountReady = ("charlie.tmms." + uniqueId).equals(TEST_STUDENT2_ACCOUNT);
+        boolean isInstructorAccountReady = ("teammates.coord." + uniqueId).equals(TEST_INSTRUCTOR_ACCOUNT);
+        boolean isAdminAccountReady = ("yourGoogleId." + uniqueId).equals(TEST_ADMIN_ACCOUNT);
+        return isSecondStudentAccountReady && isInstructorAccountReady && isAdminAccountReady;
```
Comment (kanghj) [inline src/test/java/teammates/test/driver/TestProperties.java]: <code>
return TEST_STUDENT2_ACCOUNT.equals("charlie.tmms." + uniqueId) 
            && TEST_INSTRUCTOR_ACCOUNT.equals("teammates.coord." + uniqueId) 
            && TEST_ADMIN_ACCOUNT.equals("yourGoogleId." + uniqueId);
</code>

Checkstyle will complain of `String literal expressions should be on the left side of an equals comparison.`

<code>
 return ("charlie.tmms." + uniqueId).equals(TEST_STUDENT2_ACCOUNT) && ("teammates.coord." + uniqueId).equals(TEST_INSTRUCTOR_ACCOUNT) && ("yourGoogleId." + uniqueId).equals(TEST_ADMIN_ACCOUNT);
</code>

PMD complains of `UselessParentheses`, although the parenthesis are necessary.

File: src/test/java/teammates/test/driver/TestProperties.java (hunk 3 above)
Comment (kanghj) [inline src/test/java/teammates/test/driver/TestProperties.java]: Added

**Comment by wkurniawan07** [comment]
@kanghj you should merge asap while the queue is empty :p

**Pull request merged**